state/relation.go

```diff
@@ -427,10 +427,16 @@ func (r *Relation) destroyOps(ignoreApplication string, op *ForcedOperation) (op
 		}
 		return removeOps, true, nil
 	}
+	lifeAssert := isAliveDoc
+	if op.Force {
+		// Since we are force destroying, life assert should be current relation's life.
+		lifeAssert = bson.D{{"life", r.doc.Life}}
+	}
+
 	return []txn.Op{{
 		C:      relationsC,
 		Id:     r.doc.DocID,
-		Assert: bson.D{{"life", Alive}, {"unitcount", bson.D{{"$gt", 0}}}},
+		Assert: append(bson.D{{"unitcount", bson.D{{"$gt", 0}}}}, lifeAssert...),
```

Comment:
append(lifeAssert, {"unitcount", bson.D{{"$gt", 0}}})

Reply:
I am pretty sure either way will work and will achieve the same result but I'll change :)

Reply:
Actually the reverse does not work as you'd get an error similar to "cannot use bson.D{...} as type DocElement"

The correct syntax, if you wanted to change the order would have been "append(lifeAssert, bson.D{{"unitcount", bson.D{{"$gt", 0}}}}...)" which does not give you much of a difference from what I have written.

The only way that it could work is if lifeAssert was declared as bson.DocElement but it would not work because our isAliveDoc is bson.D already.

So no change is needed here.